src/Components/Server/src/Circuits/CircuitInboundEventDelegate.cs

```diff
@@ -0,0 +1,11 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components.Server.Circuits;
+
+/// <summary>
+/// A function that handles an inbound <see cref="Circuit"/> event.
+/// </summary>
+/// <param name="context">The <see cref="CircuitInboundEventContext"/> for the event.</param>
+/// <returns>A <see cref="Task"/> that completes when the event has finished.</returns>
+public delegate Task CircuitInboundEventDelegate(in CircuitInboundEventContext context);
```

Comment:
I would avoid this declaration and directly use `Func<Context,Task>`

Reply:
That means we can't make `CircuitInboundEventContext` an `in` parameter, is that ok?

Reply:
Never mind, just saw your comment about making `CircuitInboundEventContext` a class.